st2api/st2api/controllers/v1/policies.py

```diff
@@ -0,0 +1,263 @@
+        return resource_db
+
+    def _get_filters(self, **kwargs):
+        filters = copy.deepcopy(kwargs)
+        ref = filters.get('ref', None)
+
+        if ref:
+            try:
+                ref_obj = PolicyTypeReference.from_string_reference(ref=ref)
+            except InvalidReferenceError:
+                raise
+
+            filters['name'] = ref_obj.name
+            filters['resource_type'] = ref_obj.resource_type
+            del filters['ref']
+
+        return filters
+
+
+class PolicyController(resource.ContentPackResourceController):
+    model = PolicyAPI
+    access = Policy
+
+    supported_filters = {
+        'pack': 'pack'
+    }
+
+    query_options = {
+        'sort': ['pack', 'name']
+    }
+
+    @jsexpose(body_cls=PolicyAPI, status_code=http_client.CREATED)
+    def post(self, instance):
+        """
+            Create a new policy.
+            Handles requests:
+                POST /policies/
+        """
+        op = 'POST /policies/'
+
+        if not hasattr(instance, 'enabled'):
+            LOG.debug('%s default unset "enabled" to True.', op)
+            setattr(instance, 'enabled', True)
+        else:
+            instance.enabled = bool(instance.enabled)
+
+        if not hasattr(instance, 'pack'):
+            LOG.debug('%s default unset "pack" to %s.', op, DEFAULT_PACK_NAME)
+            setattr(instance, 'pack', DEFAULT_PACK_NAME)
+
+        db_model = self.model.to_model(instance)
+        LOG.debug('%s verified object: %s', op, db_model)
+
+        try:
+            db_model = self.access.add_or_update(db_model)
+        except StackStormDBObjectConflictError as e:
+            # If an existing DB object conflicts with new object then raise error.
+            LOG.exception('%s unable to create object due to uniqueness '
+                          'conflict: %s', op, db_model)
+            abort(http_client.CONFLICT, str(e), body={'conflict-id': e.conflict_id})
+            return
+        except Exception as e:
+            LOG.exception('%s unable to create object: %s', op, db_model)
+            abort(http_client.INTERNAL_SERVER_ERROR, str(e))
+            return
+
+        LOG.debug('%s created object: %s', op, db_model)
+        LOG.audit('Policy created. Policy.id=%s' % (db_model.id), extra={'policy_db': db_model})
+
+        return self.model.from_model(db_model)
+
+    @jsexpose(arg_types=[str], body_cls=PolicyAPI)
+    def put(self, ref_or_id, instance):
+        op = 'PUT /policies/%s/' % ref_or_id
+
+        try:
+            db_model = self._get_by_ref_or_id(ref_or_id=ref_or_id)
+            LOG.debug('%s found object: %s', op, db_model)
+        except Exception as e:
+            LOG.exception('%s unable to find object.', op)
+            abort(http_client.NOT_FOUND, e.message)
+            return
+
+        db_model_id = db_model.id
+
+        try:
+            validate_not_part_of_system_pack(db_model)
```

Comment:
I copied this from the ContentPack controller.  I guess I can use cases where users may want to either delete and change the policies that are deployed as part of the core pack.  But it is going to be inconsistent with how resources (i.e. actions) are managed under the core pack.  Are we ok with that?